### PR TITLE
Update compat for ClimaCoreTempestRemap

### PR DESCRIPTION
**Pull request by sriharshakandala**
- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: lib/ClimaCoreTempestRemap/Project.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name = "ClimaCoreTempestRemap"
 uuid = "d934ef94-cdd4-4710-83d6-720549644b70"
 authors = ["CliMA Contributors <clima-software@caltech.edu>"]
-version = "0.3.7"
+version = "0.3.8"
```
Comment (simonbyrne) [inline lib/ClimaCoreTempestRemap/Project.toml:5]: We never actually tagged 0.3.7

Comment (simonbyrne) [reply lib/ClimaCoreTempestRemap/Project.toml:5]: So you don't need to update this

Comment (sriharshakandala) [reply lib/ClimaCoreTempestRemap/Project.toml:5]: I seems to be tagged here: https://github.com/CliMA/ClimaCore.jl/commit/e5a2ec9c89f835198c25e44fe8c578c2f439b427#diff-1fa6758b8a92bc430e50639e6151ba4f28598df076aee726a46f8c821e86acce
Not sure why this wasn't in the registry!

Comment (simonbyrne) [reply lib/ClimaCoreTempestRemap/Project.toml:5]: looks like registrator didn't pick it up

Just up the min version, then we can tag it

**Pull request merged**